### ISSUE TITLE
Cannot publish to a self-hosted Wordpress site that is using HTTPS

### ISSUE THREAD

**Issue by DDReaper** [OpenLiveWriter/OpenLiveWriter#580]
I recently updated my self hosted wordpress site to use HTTPS.

However now, I cannot use or publish to the site using openlivewriter :(

It took me a few attempts to even get the site registered, but it worked eventually with a few twists and turns.  but when I came to publish, this was all for naught.

Erm, help?

**Comment by ScottIsAFool** [comment]
What errors are you getting?

**Comment by DDReaper** [comment]
If I publish I get (which I suppose is expected)
![image](https://cloud.githubusercontent.com/assets/1793042/23773817/cf578ff2-0518-11e7-91ac-373abfff3e74.png)

Going back to edit my Blog config I get the following (which is pretty much what I got initially)
![image](https://cloud.githubusercontent.com/assets/1793042/23773860/fcb2066c-0518-11e7-957d-663e95207145.png)

Seems not to support https connections?

**Comment by ScottIsAFool** [comment]
Are you behind a proxy at all? I've just tried adding your blog (albeit with useless credentials) and it goes through the normal motions and only fails at the point of credentials. I'm assuming what you're seeing is a near instant failure where it doesn't try and get blog publishing interface, etc?

**Comment by DDReaper** [comment]
Nope, a direct connection.
I also noted that it couldn't detect it as a Wordpress site on the initial setup.  When you tested @ScottIsAFool , Did it detect or ask you what type of site it was?

Always recognised it before I updated to HTTPS

**Comment by DDReaper** [comment]
What is very odd. Is that if I select "HTTP" as the address, the blog detection screen appears.  However it skips it when I correctly select "HTTPS".
I'm checking through my site to make sure it's not an issue there.

**Comment by DDReaper** [comment]
After a lot of hacking (and bringing down) my site, I may have an answer for why it may not be working (and it's probably not me)
https://wordpress.org/support/topic/403-forbidden-on-xmlrpcphp/

According to Wordpress, hosting org are blocking access to xmlrpc which OpenLiveWriter depends on for Wordpress.  If so, it means it is likely NOT to work in most places.

Maybe time for another plan for OLW?  I really like OLW and want it back badly 🗡

**Comment by ScottIsAFool** [comment]
Unfortunately, if the hosting companies themselves are blocking that, there's not a lot OLW can do. Unless wordpress suddenly start offering some kind of oauth approach, we will have to stick with what we have, but at least we have something that we can get people to check with their hosting company.

In the meantime, it might be worth speaking with your hosting provider and find out if this is indeed the case and what they can do about it.

**Comment by ScottIsAFool** [comment]
@DDReaper Are you still having this issue? If so, would you be willing to try a test version when one is ready as I don't have a blog with which to try this.

**Comment by DDReaper** [comment]
Always ready to test anything @ScottIsAFool .
Have a few spare sites to test most blogs / sites :D

**Comment by DDReaper** [comment]
The situation continues. They did find a block on an IP with the site and unblocked access.
Browsing to the xmlrpc.php file now reports "only post requests allowed"
However, OLW still can't access it :S

**Comment by DDReaper** [comment]
Hmm, interesting.
I chanced it and selected http instead of https as the url and it actually worked? and posted.  
So I have a "sort of" work around now", but if you have a newer experimental way not using xmlrpc @ScottIsAFool  I'm willing to try it, considering I'm not sure how long this will stay open.

**Comment by ScottIsAFool** [comment]
We want to check that it's nothing to do with OLW not supporting SSL, and we also want to see if it's anything to do with redirects, because i note the http link of your site does redirect to the https version. Won't have anything to test until later this weekend at the earliest though.

**Comment by DDReaper** [comment]
OK, no worries.  I'll be ready when you are.

**Comment by ScottIsAFool** [comment]
@DDReaper Could you please try installing this version https://ci.appveyor.com/project/dotnetfoundation/openlivewriter/build/0.6.2.287-sulpsasv/artifacts and seeing whether you can connect to your https blog now, please.

**Comment by DDReaper** [comment]
That worked a treat first time. Detected the blog and published successfully.
I take it that it's still using the xmlrpc route currently instead of the Wordpress/Jetpack API's.

**Comment by ScottIsAFool** [comment]
You worry me saying "first time", I know what you mean, but just to be safe, can you try adding it again :)

**Comment by DDReaper** [comment]
I've uninstalled the test version you sent twice now.
Each time, it detects the blog properly (as you would expect) creates the temporary post (and deletes it)
I then post a draft post and it works fine.

So it is now working as expected, as it did before I updated to HTTPs

**Comment by ScottIsAFool** [comment]
Awesome, and just to be a pain (sorry), if you install the original version (from openlivewriter.org), you then can't add it, right? Just want to confirm the original problem can still be reproduced with the original version.

**Comment by DDReaper** [comment]
Sure, give me two ticks.  (and I know what you mean about a fix working "first time") it's also a "first " for me :P

**Comment by DDReaper** [comment]
Confirmed. Re-installing the current production version fails as before:
![image](https://cloud.githubusercontent.com/assets/1793042/23897789/5e20c8da-08a6-11e7-80e5-d5134500ec19.png)

**Comment by ScottIsAFool** [comment]
Sweet. Thanks for testing this :)

**Comment by DDReaper** [comment]
No worries, in my best interest to help where I can.
Not in a position to contribute code at the mo, but I can help with testing.  Azure is also good for spinning up quick test hosts (thanks to our MVP credits)

**Comment by ScottIsAFool** [comment]
Yeah, having some decent test hosts might be something @jongalloway can look into :)

**Comment by DDReaper** [comment]
True, if @jongalloway can sort out some "free" azure web hosting (it is possible to host workpress for free on Azure websites with mysql) and others like Ghost only need webspace.

**Comment by Laronski** [comment]
@ScottIsAFool This is slightly off topic. I loaded the test version you offered, just because, I was going to use it to post to **Blogger** and it wouldn't allow me to; it said _errors were encountered_— Do I need to work with you on a test there or should I create a case for it? Thoughts?

**Comment by ScottIsAFool** [comment]
What errors? Post with pictures? Does it fail on the main version (ie, not this test one)?

**Comment by Laronski** [comment]
@ScottIsAFool I will create a new thread, as this is for Open Live Writer and Blogger.

**Comment by DDReaper** [comment]
Fair enough. You happy for me to close this issue @ScottIsAFool  or do you want to wait until your PR is up?

**Comment by ScottIsAFool** [comment]
Keep it open until the PR is merged.

**Comment by jongalloway** [comment]
@ScottIsAFool @DDReaper I'm sure I can get some Azure hosting for test blogs set up, both for manual and automated testing. 

It'll take a little bit of time to set up, though and I'm pretty busy trying to wrap up the release. We should log another issue for that and assign to me, okay?

**Comment by odedyrz** [comment]
Hi,

I know this is old... I hope someone will read this and know the answer. 

I'm having the same issue - upgraded my website to **https**, and since then Open Live Writer and Windows Live Writer (even checked BlogDesk) can't publish - I get the **403 - Forbidden** error. The host support and I tried everything I could find in forum threads regarding this, but none worked. It seems that you guys may know what to do.

Thanks!

Already tried:
1. Check the website firewall to see it’s not blocking access to xmlrpc.php file
2. Delete cache and cookies
3. Check Folders and file permissions - 755 for folders, 644 for files. 
4. Delete .htaccess and re-create it using WP Admin
5. Rename xmlrpc.php
6. Relocate xmlrpc.php
7. Edit website urls inside wp-config.php
8. Add 3 code lines to .htaccess
9. Disable all plugins

**Comment by JoniWebb** [comment]
I had this problem.   The download is the problem.  Please use this download, it should work.  It did for me.  https://www.mybloggertricks.com/2015/12/fixing-open-live-writer-errors.html

My live write just stopped for no reason at all.  Did all you did.  Then found this download and tried it.  It worked for me.

Let us know if this lets you onto live writer.  thanks!
Joni

**Comment by odedyrz** [comment]
Thanks, but same result...

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
